_pytest/main.py

```diff
@@ -90,10 +90,12 @@ def wrap_session(config, doit):
             session.exitstatus = doit(config, session) or 0
         except pytest.UsageError:
             raise
-        except KeyboardInterrupt:
+        except KeyboardInterrupt as e:
             excinfo = _pytest._code.ExceptionInfo()
             config.hook.pytest_keyboard_interrupt(excinfo=excinfo)
             session.exitstatus = EXIT_INTERRUPTED
+        except pytest.Exit as e:
+            sys.stderr.write('{0}: {1}\n'.format(type(e).__name__, e.msg))
```

Comment:
What is the session.exitstatus when pytest.Exit is caught? It looks like it might be 0 and before this change it was EXIT_INTERNALERROR. But I might be wrong... maybe check it in the test?